### ISSUE TITLE
Variadic macro expansion

### ISSUE THREAD

**Issue by MKHen** [doxygen/doxygen#8542]
I want to document a macro like

`HELPS(enumName, uint8_t, element1 = 4, element2, last);`
with Doxygen and convert it with the preprocessor to

```
enum class enumName
{
   element1 = 4,
   element2,
   last
};
```
to get a common enum documentation. I used already the PREDEFINED tag in doxygen with

`HELPS(param1,param2,...)="enum class param1 { __VA_ARGS__ }"`
but this won't work. Doxygen version is 1.9.1 Language is C++.

I usually document enum classes like
```
/**
 *  @enum enumName
 *  @brief a enum test class
 *  @var enumName::element1
 *  It is 4
 *  @var enumName::element2
 *  It is 5
 *  @var enumName::last
 *  It is 6
 */
enum class enumName
{
    element1 = 4,
    element2,
    last
};
```

which works pretty well. For some reasons I am now forced to use a Macro (HELPS) and not a enum. The behaviour is the same and the documentation shoulb be as well.

I wrote:

```
/**
 *  @enum enumName2
 *  @brief a enum test class
 *  @var enumName2::element10
 *  It is 40
 *  @var enumName2::element20
 *  It is 41
 *  @var enumName2::last
 *  It is 42
 */
HELPS(enumName2, uint8_t, element10 = 40, element20, last);
```

which should expand in the doxygen preprocessor to the good working example aboth but sadly (with -d Preprocessor) it expands to:

```
/**
 *  @enum enumName2
 *  @brief a enum test class
 *  @var enumName2::element10
 *  It is 40
 *  @var enumName2::element20
 *  It is 41
 *  @var enumName2::last
 *  It is 42
 */
element10 = 40   class   enumName2   { __VA_ARGS__ } element20, last);
```

which results in a bad documentation.

Any suggestions?
Please observe the original post on [stackoverflow](https://stackoverflow.com/questions/67472449/document-special-macro-with-doxygen) as well.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8544

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release) with a small refactoring (and a small comment (https://github.com/doxygen/doxygen/commit/c53818345c516165fe2502ebdba9017c97f0bd99#commitcomment-51115694) regarding the fact that in the used `PREDEFINED` the `...` have to be at the end.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by MKHen** [comment]
I tested it.
Working like expected now.
Thanks you for the good work.
